### PR TITLE
Use -isystem instead of -I for hipSYCL headers to avoid warnings with high warning levels

### PR DESCRIPTION
**Pull request by illuhad**
@krasznaa Please check if this fixes the warning issue for you.

### CODE REVIEW

File: bin/syclcc-clang [hunk 1]
```diff
@@ -1665,7 +1665,7 @@ class compiler:
   @property
   def common_cxx_flags(self):
     args = [
-      "-I" + self._hipsycl_include_path,
+      "-isystem" + self._hipsycl_include_path,
```
Comment (krasznaa) [inline bin/syclcc-clang]: I thought that one would have to leave a space between `-isystem` and the path. :confused: But this does seem to work as-is. :+1: At least in my own test the warnings are gone with this included.

**Review by krasznaa — APPROVED** [review]
Ahh, didn't realise that I could even approve this PR. :stuck_out_tongue: (And it's good that I can't... I thought I may have found a mistake in the repository's setup.)

File: bin/syclcc-clang (hunk 1 above)
Comment (illuhad) [reply bin/syclcc-clang]: hm.. good catch, I thought so too :P

Comment (krasznaa) [reply bin/syclcc-clang]: Just to confirm that the updated/fixed formalism also works as desired in my test.

Comment (illuhad) [reply bin/syclcc-clang]: Thanks! :)

**Pull request merged**